Retrie/Types.hs

```diff
@@ -45,6 +45,7 @@ module Retrie.Types
   , RewriterResult(..)
   , ParentPrec(..)
   , Context(..)
+  , Extension(..)
```

Comment:
Re-exporting the type here to reduce the noise associated with importing the GHC module.